qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -292,67 +324,153 @@ def circuits(self) -> List[QuantumCircuit]:
             A list of :class:`QuantumCircuit`.
 
         Raises:
-            AttributeError: When the backend doesn't report the time resolution of waveforms.
+            QiskitError: When the backend is not set with use of pulse gate.
```

Comment:
Use of pulse gate (CRPulseGate) matters here because we should take channel mapping from the backend to build schedule to attach. If cr gate is a unitary, e.g. Hamiltonian gate, then missing backend should be okey because this doesn't generate schedule. Note that we can still submit a pulse gate job to backend with open pulse disabled.

I updated error message in https://github.com/Qiskit/qiskit-experiments/pull/933/commits/4edabed1c60757001b292ba07d719b0b0d2ff74f

Reply:
I see what you are trying to check for. Could the user not easily bypass this check by providing some random gate object? E.g. do `cr_gate=Gate("my_gate", 2, [])`